docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/ContainerCache.java

```diff
@@ -15,22 +15,33 @@
  */
 package com.palantir.docker.compose.connection;
 
+import static java.util.stream.Collectors.toSet;
+
+import com.palantir.docker.compose.execution.Docker;
 import com.palantir.docker.compose.execution.DockerCompose;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
 public class ContainerCache {
 
     private final Map<String, Container> containers = new HashMap<>();
+    private final Docker docker;
     private final DockerCompose dockerCompose;
 
-    public ContainerCache(DockerCompose dockerCompose) {
+    public ContainerCache(Docker docker, DockerCompose dockerCompose) {
+        this.docker = docker;
         this.dockerCompose = dockerCompose;
     }
 
     public Container container(String containerName) {
-        containers.putIfAbsent(containerName, dockerCompose.container(containerName));
+        containers.putIfAbsent(containerName, new Container(containerName, docker, dockerCompose));
```

Comment:
All this wiring in of `docker`, `docker-compose` dependencies feels a bit weird - not sure I have a better idea though